ras_backstage/controllers/secure_messaging_controller.py

```diff
@@ -43,6 +43,27 @@ def get_message(encoded_jwt, message_id, is_draft):
     return json.loads(response.text)
 
 
+def get_thread_by_id(encoded_jwt, thread_id):
+    """
+    The method return all the message part of a thread.
+    :param encoded_jwt:  is the security token.
+    :param thread_id: is the id of the thread selected.
+    :return: a list of messages sent/received part of conversation thread
+    """
+
+    url = f"{app.config['RAS_SECURE_MESSAGING_SERVICE_THREAD']}{thread_id}"
+
+    headers = _create_authorization_header(encoded_jwt)
+    response = request_handler('GET', url, headers=headers)
+
+    if response.status_code != 200:
+        logger.error('Error retrieving the messages', thread_id=thread_id)
+        raise ApiError(url, response.status_code)
+
+    logger.debug('Successfully retrieved message', thread_id=thread_id)
+    return json.loads(response.text)
```

Comment:
```Python
return response.json()
```

also, you might want to guard against ValueErrors when decoding arbitrary JSON.

Reply:
If it does have an impact might be worth adding tech debt for this as it's in multiple different places?

Reply:
The impact would be that the whole process would exit with a non-zero code when the ValueError is raised (I do not think this is caught elsewhere). If it is in multiple places, then we need to evaluate the risk and decide whether to guard against this happening (which it eventually will).

Reply:
If we do decide to fix it elsewhere, that doesn't mean allowing this to get in to the codebase is ok.

Reply:
done it.